### DISCUSSION TITLE
Shell isn't interactive

### DISCUSSION THREAD

**Discussion by RickyBubbles** [filebrowser/filebrowser#984] [Ideas]
**Description**
unable to run ssh in the shell because of this error:
Pseudo-terminal will not be allocated because stdin is not a terminal.

**Expected behaviour**
What did you expect to happen?
I expected to be able to use ssh and/or IRC in an interactive fashion

**What is happening instead?**
ssh connected and then terminates with the error:
Pseudo-terminal will not be allocated because stdin is not a terminal.

**Additional context**
No additional context... just want to use the shell for things that are interactive like ssh or IRC, etc.

**How to reproduce?**
Go ssh in to something.

**Comment by StefVanG** [comment]
I've the same problem: after the positive resolution of the issue #982 I've recompiled Filebrowser in an Alpine container with ssh to manage docker containers directly from the Filebrowser's shell.

I see the correct ssh banner but after that no prompt is proposed and the error is "Pseudo-terminal will not be allocated because stdin is not a terminal".

I've tried different flags on the ssh command but none work.

I hope @o1egl will get rid of this problem now that is labelled as a feature because it'll be a very powerful feature!

**Reply by samsalfer** [reply]
hello @StefVanG can you share the docker-compose.yml with this solution? 
thanks

**Comment by RickyBubbles** [comment]
My hope was to be able to turn it in to a chatbox for active users. Obviously my ability to do this would depend on the way this was implemented... but if anyone ever decides to make this a feature, please do it in a way that allows communication between logged in users (maybe via a local IRC server or some sort of ncurses based chat program or the like)

**Comment by michacassola** [comment]
Yes, if there was a real interactive shell, and preferrably one that can be extended to the whole screen this would have been the perfect tool for me!
It should run as a sudo user and with /bin/bash.
I highly suggest xterm.js: https://github.com/xtermjs/xterm.js
